src/main/java/org/junit$pioneer/jupiter/TempDirectory.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit$pioneer.jupiter;
+
+import java.io.IOException;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+/**
+ * {@code TempDirectory} is a JUnit Jupiter extension to create and cleanup a
+ * per-test temporary directory.
+ *
+ * <p>The temporary directory is only created if a test or lifecycle method has
+ * a parameter annotated with the {@link Temp} annotation. If the parameter type
+ * is not {@link Path} or when the temporary directory could not be created,
+ * this extension will throw a {@link ParameterResolutionException}.
+ *
+ * <p>After the test is finished, this extension will attempt to recursively
+ * delete all directories and files in the temporary directory and, finally, the
+ * temporary directory itself. In case deletion of a file or directory fails,
+ * this extension will throw an {@link IOException} that will cause the test to
+ * fail.
+ *
+ * @since 0.1
+ * @see Temp
+ */
+public class TempDirectory implements ParameterResolver {
+
+	/**
+	 * {@code Temp} can be used to annotate a test or lifecycle method parameter
+	 * of type {@link Path} that requires a temporary directory.
+	 *
+	 * @see TempDirectory
+	 */
+	@Target(ElementType.PARAMETER)
+	@Retention(RetentionPolicy.RUNTIME)
+	@Documented
+	public @interface Temp {
+	}
+
+	private static final Namespace NAMESPACE = Namespace.create(TempDirectory.class);
+	private static final String KEY = "dir";
+
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		return parameterContext.isAnnotated(Temp.class);
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext context) {
+		Class<?> parameterType = parameterContext.getParameter().getType();
+		if (!Path.class.equals(parameterType)) {
```

Comment:
I think `!=` should suffice since you're comparing to a core Java type.

Reply:
ok, will change that